packages/kbn-router-to-openapispec/src/oas_converter/zod/lib.test.ts

```diff
@@ -0,0 +1,167 @@
+                items: [
+                  {
+                    type: 'string',
+                  },
+                  {
+                    type: 'string',
+                  },
+                ],
+                maxItems: 2,
+                minItems: 2,
+                type: 'array',
+              },
+              maxItems: 125,
+              type: 'array',
+            },
+            maybeNumber: {
+              maximum: 1000,
+              minimum: 1,
+              type: 'number',
+            },
+            neverType: {
+              not: {},
+            },
+            record: {
+              additionalProperties: {
+                type: 'string',
+              },
+              type: 'object',
+            },
+            string: {
+              maxLength: 10,
+              minLength: 1,
+              type: 'string',
+            },
+            union: {
+              anyOf: [
+                {
+                  description: 'Union string',
+                  maxLength: 1,
+                  type: 'string',
+                },
+                {
+                  description: 'Union number',
+                  minimum: 0,
+                  type: 'number',
+                },
+              ],
+            },
+            uri: {
+              default: 'prototest://something',
+              format: 'uri',
+              type: 'string',
+            },
+          },
+          required: ['string', 'ipType', 'literalType', 'neverType', 'map', 'record', 'union'],
+          type: 'object',
+        },
+        shared: {},
+      });
+    });
+  });
+
+  describe('convertPathParameters', () => {
+    test('base conversion', () => {
+      expect(
+        convertPathParameters(z.object({ a: z.string() }), { a: { optional: false } })
+      ).toEqual({
+        params: [
+          {
+            in: 'path',
+            name: 'a',
+            required: true,
+            schema: {
+              type: 'string',
+            },
+          },
+        ],
+        shared: {},
+      });
+    });
+    test('throws if known parameters not found', () => {
+      expect(() =>
+        convertPathParameters(z.object({ b: z.string() }), { a: { optional: false } })
+      ).toThrow(
+        'Path expects key "a" from schema but it was not found. Existing schema keys are: b'
+      );
+    });
+  });
+
+  describe('convertQuery', () => {
+    test('base conversion', () => {
+      expect(convertQuery(z.object({ a: z.string() }))).toEqual({
+        query: [
+          {
+            in: 'query',
+            name: 'a',
+            required: true,
+            schema: {
+              type: 'string',
+            },
+          },
+        ],
+        shared: {},
+      });
+    });
+  });
+
+  describe('is', () => {
+    test.each([
+      [{}, false],
+      [1, false],
+      [undefined, false],
+      [null, false],
+      [z.any(), true],
+      [z.object({}).default({}), true],
+      [z.never(), true],
```

Comment:
Any, object, and never were `false` in the config-schema test, but I changed it to `true` here.

Reply:
Sure! If it can handle those correctly, that makes sense!